### PR TITLE
Support server-per-port deployment

### PR DESCRIPTION
**Pull request by dbishop**
Support server-per-port deployment

### CODE REVIEW

File: cookbooks/swift/recipes/configs.rb [hunk 1]
```diff
@@ -134,6 +138,19 @@
     end
   end
   (1..node['nodes']).each do |i|
+    bind_ip = "127.0.0.1"
+    bind_port = "60#{i}#{p}"
+    if p == 0 && node['servers_per_port'] > 0 then
```
Comment (clayg) [inline cookbooks/swift/recipes/configs.rb]: p == 0 is a correct, but non-obvious way to spell "this is the object service", maybe:

    if service == "object" && ... then
      ...

File: cookbooks/swift/recipes/rings.rb [hunk 2]
```diff
@@ -62,14 +65,25 @@
...
+    dev = "sdb#{i}"
+    ip = "127.0.0.1"
+    port = "60#{n_idx}0"
+    if node['servers_per_port'] > 0 then
+      ip = "127.0.0.#{n_idx}"
+
+      # Range ports per disk per node from 60j6 - 60j9
+      # NOTE: this only supports DISKS <= 4 * NODES
+      p = 5 + (i / Float(node['nodes'])).ceil.to_int
+      port = "60#{n_idx}#{p}"
```
Comment (clayg) [inline cookbooks/swift/recipes/rings.rb]: so right now we have account & container's on ports `60#{n_idx}1` and `60#{n_idx}2` - in the non per-port case objects use `60#{n_idx}0`.

Maybe in the next go ground if  `servers_per_port > 0` - objects should just move to `"6#{n_idx}%02d" % i` - should work up to 99 disks and keep sdbXX matching with 6?XX.

File: localrc-template [hunk 3]
```diff
@@ -1,10 +1,11 @@
 export VAGRANT_BOX=swift-all-in-one  # trusty, precise, etc.
 export VAGRANT_HOSTNAME=saio
-export VAGRANT_RAM=768
+export VAGRANT_CPUS=1
+export VAGRANT_RAM=1024
```
Comment (clayg) [inline localrc-template]: Should change the real default in the Vagrantfile

File: cookbooks/swift/templates/default/etc/swift/object-server/default.conf-template.erb [hunk 4]
```diff
@@ -3,6 +3,8 @@ mount_check = false
 disable_fallocate = true
 workers = 1
 log_facility = LOG_LOCAL2
+# NOTE: servers_per_port necessary for replicator/reconstructor so must go here
+servers_per_port = <%= @servers_per_port %>
```
Comment (clayg) [inline cookbooks/swift/templates/default/etc/swift/object-server/default.conf-template.erb]: I don't yet understand why replicator/reconstructor need this - I get why they might need bind_ip...

Comment (dbishop) [inline cookbooks/swift/templates/default/etc/swift/object-server/default.conf-template.erb]: Line 70 (on the right of the diff?) at
https://review.openstack.org/#/c/184189/13/swift/obj/replicator.py

Comment (clayg) [inline cookbooks/swift/templates/default/etc/swift/object-server/default.conf-template.erb]: Well... that's not really what I meant.  I mean I just need to finish loading my brain.

Like half the time I can't even remember why the replicator needs to have a *port* much less the bind_ip and the servers_per_port integer...  I assume it has something to do with "how the fuck do i know if this thing in the ring is mine" - which IIRC is harder on a swift-all-in-one than on a real server with real ips - but I'm sure I'll be able to figure it out.

So this was really just a note to myself that I might be able to improve the comment here after I understand the *why*

File: cookbooks/swift/recipes/rings.rb (hunk 2 above)
Comment (dbishop) [inline cookbooks/swift/recipes/rings.rb]: Maybe... I know at least one place (a probe test?) was doing wacky shit like examining port number % 10 or % 100 or something to map back to node number or ring type... can't remember exactly.  But that's the sort of shit any change to these port numbers has a chance to break.

But it's be easy to test if your proposed change hurts any of the probe tests.

File: localrc-template (hunk 3 above)
Comment (dbishop) [inline localrc-template]: fixed

**Pull request merged**